### DISCUSSION TITLE
How to draw a point on a line that will be at a certain distance from the starting point?

### DISCUSSION THREAD

**Discussion by deomid-fedorov** [openlayers/openlayers#16322] [Q&A]
I have a line that consists of five points. How can I draw a point on this line that will be at a certain distance from the starting point? This distance will not be greater than the length of the entire line.

**Comment by ShelleyLake** [comment]
What about using "turf.js"  - function "turf.along" ?
Have a look at https://openlayers.org/en/latest/examples/turf.html


```
// Define your line with five points
const line = turf.lineString([
  [0, 0],
  [1, 1],
  [2, 2],
  [3, 3],
  [4, 4]
]);

// Define the distance from the starting point (in kilometers)
const distance = 2; // Adjust this value as needed

// Get the point at the specified distance along the line
const point = turf.along(line, distance, { units: 'kilometers' });

// Extract the coordinates from the point
const coordinates = point.geometry.coordinates;

// Draw something in OL
...
```

**Comment by mike-000** [comment]
Use one of the `getLength` methods (either https://openlayers.org/en/latest/apidoc/module-ol_sphere.html#.getLength or https://openlayers.org/en/latest/apidoc/module-ol_geom_LineString-LineString.html#getLength depending on your requirement) to calculate the total length, calculate the fraction of that then use the `getCoordinateAt` method to return the point on the line as in https://openlayers.org/en/latest/examples/feature-move-animation.html
